File: Main2_43.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class Main2_43 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        Set<String> dict = new HashSet<>();

        for (int i = 0; i < n; i++) {
            String command = sc.next();
            String str = sc.next();

            if (command.equals("insert")) {
                dict.add(str);
            } else if (command.equals("find")) {
                if (dict.contains(str)) {
                    System.out.println("yes");
                } else {
                    System.out.println("no");
                }
            }
        }

        sc.close();
    }
}

File: Main2_43.py
def main():
    n = int(input())
    dictionary = set()

    for _ in range(n):
        command, string = input().split()

        if command == "insert":
            dictionary.add(string)
        elif command == "find":
            if string in dictionary:
                print("yes")
            else:
                print("no")


if __name__ == "__main__":
    main()